mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -793,13 +799,64 @@ public void forEachTuple(ModelConsumer consumer, EntityKeyMetadata... entityKeyM
 		String collectionName = getCollectionName( backendQuery, queryDescriptor, entityKeyMetadata );
 		DBCollection collection = provider.getDatabase().getCollection( collectionName );
 
+		// Recap, Section 3.8.15 of the JPA specification states: "The use of named parameters is not defined for
+		// native queries. Only positional parameter binding for SQL queries may be used by portable applications."
+		// This should apply analogously to MongoDB native queries.
+//		if ( !queryParameters.getNamedParameters().isEmpty() )
+//			throw new IllegalArgumentException( "The use of named parameters is not defined for native queries; c.f. JPA spec, Section 3.8.15." );
+		if ( !queryParameters.getPositionalParameters().isEmpty() ) // TODO Implement binding positional parameters.
+			throw new UnsupportedOperationException( "Positional parameters are not yet supported for MongoDB native queries." );
+
 		switch( queryDescriptor.getOperation() ) {
 			case FIND:
 				return doFind( queryDescriptor, queryParameters, collection, entityKeyMetadata );
+			case FINDONE:
+				return doFindOne( queryDescriptor, collection, entityKeyMetadata );
+			case FINDANDMODIFY:
+				return doFindAndModify( queryDescriptor, collection, entityKeyMetadata );
 			case AGGREGATE:
 				return doAggregate( queryDescriptor, queryParameters, collection, entityKeyMetadata );
 			case COUNT:
 				return doCount( queryDescriptor, collection );
+			case INSERT:
+			case REMOVE:
+			case UPDATE:
+				throw new IllegalArgumentException( "Query must be executed using 'executeUpdate' method: " + queryDescriptor );
+			default:
+				throw new IllegalArgumentException( "Unexpected query operation: " + queryDescriptor );
+		}
+	}
+
+	@Override
+	public int executeBackendUpdateQuery(final BackendQuery<MongoDBQueryDescriptor> backendQuery, final QueryParameters queryParameters)
+	{
```

Comment:
Don't know what you mean here. Anyway, feel free to change it according to your style guide.